src/components/ChecSlideoutPanel/Overlay.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+  <div
+    :class="`slideout-panel__overlay slideout-panel__overlay--${overlay}`"
+    @click="onOverlayDismiss"
+  />
+</template>
+
+<script>
+export default {
+  name: 'Overlay',
+  props: {
+    /**
+     * The overlay backdrop when the panel is opened. One of "light" and "dark".
+     */
+    overlay: {
+      type: String,
+      default: 'light',
+      validator(value) {
+        return ['light', 'dark'].includes(value);
+      },
+    },
+    /**
+     * Whether to close panel on overlay click. Default is true.
+     */
+    closeOnOverlayClick: Boolean,
+  },
+  methods: {
+    onOverlayDismiss() {
+      if (this.closeOnOverlayClick) {
+        this.$emit('close');
+      }
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+.slideout-panel {
+  &__overlay {
+    @apply
+      fixed
+      top-0
+      left-0
+      w-full
+      h-full
+      flex
+      z-50
+      overflow-scroll
```

Comment:
This is a bit weird IMO. I think when a panel is open that whatever was behind is should be static. You can disable scrolling behind modals by putting `overflow: hidden` on the body when a panel/modal/etc is open. Maybe check to see whether we already do something for this in the modal, and if so we might be able to re-use that. If we don't then we probably should and in this case we can probably tackle this as a separate issue and kill two birds with one stone.